### PR TITLE
Added Interactions restrictions API for organizations

### PR DESCRIPTION
**Pull request by waseem18**
Issue : #1077

### CODE REVIEW

File: github/interactions_orgs.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 The go-github AUTHORS. All rights reserved.
```
Comment (gmlewis) [inline github/interactions_orgs.go]: Would you mind bumping the Copyright year on new files (only) to 2019, please?
All existing files can keep their original Copyright year unchanged.
Thanks!

File: github/interactions_orgs.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		return nil, resp, err
+	}
+
+	return organizationInteractions, resp, nil
+}
+
+// UpdateRestrictionsForOrg adds or updates the interaction restrictions for an organization.
+//
+// GitHub API docs: https://developer.github.com/v3/interactions/orgs/#add-or-update-interaction-restrictions-for-an-organization
+func (s *InteractionsService) UpdateRestrictionsForOrg(ctx context.Context, organization string, interaction *InteractionRestriction) (*InteractionRestriction, *Response, error) {
```
Comment (gmlewis) [inline github/interactions_orgs.go]: It just dawned on me that this API currently only takes a single parameter (`limit`) and yet we are passing in a full `InteractionRestriction` struct.

GitHub has recently been clamping down on extra parameters and is wanting to return errors when extra parameters are sent.

Therefore, we have to be extremely careful at what is being sent.

I recommend for now that we pass in another string for the required `limit` parameter and that we document what values can be passed.

Let's also go back and double-check your recent additions for repos and make sure that we are not violating this same principle, as I forgot to check that carefully.

So to be precise, I'm thinking of something like this:

```go
// UpdateRestrictionsForOrg adds or updates the interaction restrictions for an organization.
//
// limit specifies the group of GitHub users who can comment, open issues, or create pull requests
// in public repositories for the given organization.
// Possible values are: "existing_users", "contributors_only", "collaborators_only".
//
// GitHub API docs: https://developer.github.com/v3/interactions/orgs/#add-or-update-interaction-restrictions-for-an-organization
func (s *InteractionsService) UpdateRestrictionsForOrg(ctx context.Context, organization, limit string) (*InteractionRestriction, *Response, error) {
```

And then, within the function, you can still use that same struct, but we are then assured of the other fields being empty:

```go
interaction := &InteractionRestriction{limit: String(limit)}
```

File: github/interactions_orgs_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 The go-github AUTHORS. All rights reserved.
```
Comment (gmlewis) [inline github/interactions_orgs_test.go]: Same here please: s/2018/2019/

File: github/interactions_orgs.go (hunk 2 above)
Comment (waseem18) [reply github/interactions_orgs.go]: > GitHub has recently been clamping down on extra parameters and is wanting to return errors when extra parameters are sent.

Thanks, I didn't know that.

Addressed the review in both organization and repository interactions update API's.

**Comment by gmlewis** [comment]
Thank you, @gauntface for the review!
Merging.

**Pull request merged**